rir/src/utils/optional.h

```diff
@@ -0,0 +1,2160 @@
+
```

Comment:
is it really necessary to drag this in? our build times are long enough as it is...

Reply:
not if it takes too long. But I figure some sort of optional type would be useful, we already have some workarounds in existing code. Maybe a much simpler one, like

```cpp
template <typename T>
class Optional {
  T storage;
  
 public:
  const bool hasValue;

  Optional() : hasValue(false), storage() {}
  Optional(T value) : hasValue(true), storage(value) {}
  
  T value() { assert hasValue; return storage; }
}
```

Reply:
nevermind, without SSA printing at least for this branch I don't need optional.